libs/coin-modules/coin-aptos/src/__tests__/api/index.integ.test.ts

```diff
@@ -56,19 +54,20 @@ describe("createApi", () => {
   describe("combine and broadcast", () => {
     it("sign and submit the transaction using combine and broadcast", async () => {
       // arrange
-
-      // TODO: should be replaced by the craftTransaction method
       const aptosApi = new AptosAPI(settings);
-      const transaction = createFixtureTransaction();
-      transaction.recipient = receiverAddress;
-      transaction.amount = BigNumber(10);
-      transaction.options.gasUnitPrice = DEFAULT_GAS_PRICE.toString();
-      transaction.options.maxGasAmount = DEFAULT_MAX_GAS_AMOUNT.toString();
 
-      const senderAccount = createFixtureAccount({
-        freshAddress: senderAddress,
+      const hexTx = await craftTransaction(aptosApi, {
+        amount: BigInt(10),
+        recipient: receiverAddress,
+        sender: { freshAddress: senderAddress, xpub: senderPublicKey },
+        type: "send",
+        asset: {
+          type: "native",
+        },
       });
-      const tx = await buildTransaction(senderAccount, transaction, aptosApi);
+
+      const txBytes = Hex.fromHexString(hexTx).toUint8Array();
+      const tx = RawTransaction.deserialize(new Deserializer(txBytes));
```

Comment:
Do we have a way to check amount, sender, ... after deserialization ?

Reply:
Yes, we could do it...

Reply:
done.